client/src/pages/Login/components/LoginForm.tsx

```diff
@@ -0,0 +1,130 @@
+  username: yup.string().required('Username cannot be empty'),
+  password: yup.string().required('Password cannot be empty'),
+});
+
+const LoginForm = (): JSX.Element => {
+  const navigate = useNavigate();
+  const [error, setError] = React.useState(' ');
+  const { login } = useUser();
+
+  const handleLogin = async (values: LoginVals) => {
+    try {
+      const response: AxiosResponse<LoginResponse> = await loginRequest(values);
+      if (response.status === 200) {
+        login(response.data.user.id, response.data.token);
+        navigate('/home');
+      } else {
+        setError('An error occurred, try again');
+      }
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        const axiosError: AxiosError = error;
+        if (axiosError.response?.status === 400) {
+          setError('Invalid credentials');
+        } else {
+          setError('An error occurred, try again');
+        }
+      } else {
+        setError('An error occurred, try again');
+      }
+    }
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      username: '',
+      password: '',
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values: LoginVals) => {
+      setError(' ');
+      handleLogin(values);
+    },
+  });
+
+  return (
+    <form
+      onSubmit={formik.handleSubmit}
+      style={{ display: 'flex', flexDirection: 'column' }}
+    >
+      <TextField
+        required
+        id="username"
+        label="Username"
+        variant="filled"
+        size="small"
+        inputProps={{
+          style: { color: theme.palette.primary.contrastText },
+        }}
+        InputLabelProps={{
+          style: { color: theme.palette.primary.contrastText },
+        }}
+        style={{ width: 300 }}
+        value={formik.values.username}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        error={formik.touched.username && Boolean(formik.errors.username)}
+        helperText={
+          formik.touched.username && Boolean(formik.errors.username)
+            ? formik.errors.username
+            : ' '
+        }
+      />
+      <TextField
+        required
+        id="password"
+        label="Password"
+        variant="filled"
+        size="small"
+        type="password"
+        inputProps={{
+          style: { color: theme.palette.primary.contrastText },
+        }}
+        InputLabelProps={{
+          style: { color: theme.palette.primary.contrastText },
+        }}
+        style={{ width: 300 }}
+        value={formik.values.password}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        error={formik.touched.password && Boolean(formik.errors.password)}
+        helperText={
+          formik.touched.password && Boolean(formik.errors.password)
+            ? formik.errors.password
+            : ' '
+        }
+      />
+      <Typography
+        variant="body1"
+        align="center"
+        color="error"
```

Comment:
would this error color be the deafult error or the error that is on the theme pallete?

Reply:
this is the error color we set in the global theme, so the colors are consistent throughout the app